src/Rules.hs

```diff
@@ -29,8 +30,11 @@ topLevelTargets :: Rules ()
 topLevelTargets = do
     want $ Rules.Generate.inplaceLibCopyTargets
 
-    forM_ allStages $ \stage ->
-        forM_ (knownPackages \\ [rts, libffi]) $ \pkg -> action (buildLib stage pkg)
+    if stage1Only
+       then action buildStage1Only
+       else forM_ allStages $ \stage ->
+                 forM_ (knownPackages \\ [rts, libffi]) $ \pkg ->
+                      action (buildLib stage pkg)
 
 buildLib :: Stage -> Package -> Action ()
```

Comment:
This also builds programs (not only libraries), so I think the name is confusing.